### ISSUE TITLE
Parallelizable TestFixtureSource with TestFixtureData not running in parallel

### ISSUE THREAD

**Issue by LifshesOfer** [nunit/nunit#3735]
NUnit (targeting .NET 5) does not run my TestFixture in parallel instances when using TestFixtureSource:

```
public class TestFixtureCases
{
	public static IEnumerable Cases
	{
		get
		{
			yield return new TestFixtureData("string1");
			yield return new TestFixtureData("string2");
			yield return new TestFixtureData("string3");
		}
	}
}
[TestFixture]
[Parallelizable]
[TestFixtureSource(typeof(TestFixtureCases), nameof(TestFixtureCases.Cases))]
public class FixtureClass : BasicFixtureClass
{
	public FixtureClass(string fixtureCase) : base(fixtureCase)
	{ }

	[Test]
	public void test1()
	{ }
	[Test]
	public void test2()
	{ }
}
```

However, if I'm giving the FixtureCases like the below, it _does_ run in multiple instances:
```
[TestFixture("string1")]
[TestFixture("string2")]
[TestFixture("string3")]
[Parallelizable]
public class FixtureClass : BasicFixtureClass
{
	public FixtureClass(string fixtureCase) : base(fixtureCase)
	{ }

	[Test]
	public void test1()
	{ }
	[Test]
	public void test2()
	{ }
}
```

**Comment by rprouse** [comment]
I am going to mark this as Confirm so that we can determine if this is a .NET 5 problem or a general problem. I expect it is a general problem, but I want to check.

**Comment by MagnumOpusX** [comment]
I have the exact same issue on .NET4.7.1.

**Comment by mcichonqa** [comment]
I got it in .NET Core 3.1 also.

**Comment by davidoh99** [comment]
Also having this issue; using ver 3.12 as a workaround. I'm using .NET 5 and have no problems with 3.12.

**Comment by jairbubbles** [comment]
I do confirm that using version 3.12 does fix the issue.

**Comment by MattNorgren** [comment]
Ran into this in 3.13.2, and downgrading to 3.12 fixed the issue.  Targeting netcoreapp3.1.

**Comment by MattNorgren** [comment]
@rprouse Down revving to 3.12 to get around this issue is not a good solution for me.

**Comment by CeesKaas** [comment]
I've narrowed it down to https://github.com/nunit/nunit/commit/247ede8aab1f75a99218bcabbe1eae4692b99215
which seems logical that if you don't add the attributes of a class that it wouldn't be picked up for parallelizable

**Comment by agray** [comment]
Can the one line in https://github.com/nunit/nunit/commit/247ede8aab1f75a99218bcabbe1eae4692b99215 not be added back?

**Comment by jairbubbles** [comment]
Hey all, I'm proposing a fix for this: https://github.com/nunit/nunit/pull/4099

**Comment by OsirisTerje** [comment]
@jairbubbles @agray, @CeesKaas , @MattNorgren , @davidoh99 , @mcichonqa , @TrxIsVerCtrl , @LifshesOfer  Can you guys confirm this is now working for you,  the package with the fix is here:  https://www.myget.org/feed/nunit/package/nuget/NUnit/4.0.0-dev-07745

**Comment by davidoh99** [comment]
@OsirisTerje Tested 4.0.0-dev-07854 and is working for me

**Comment by d4v1dk0n7** [comment]
Any idea about when is going to be released the stable version of nunit 4.0?
Thanks!
